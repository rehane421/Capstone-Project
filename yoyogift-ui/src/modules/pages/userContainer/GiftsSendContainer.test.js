import React from "react";
import { shallow } from "../../../enzyme";
import { shallowToJson } from "enzyme-to-json";
import { GiftsSendContainer, mapStateToProps } from "./GiftsSendContainer";

let wrapper;
let props1 = {
  isLoggedIn: true,
  user: {
    email: "johndoe@gmail.com"
  },
  fetchSentCards: jest.fn()
};
let props2 = {
  isLoggedIn: false
};
let props3 = {
  isLoggedIn: true,
  sentCards: []
};
let props4 = {
  isLoggedIn: true,
  sentCards: [{ id: 1 }]
};
it("should render GiftsSendContainer component", () => {
  wrapper = shallow(<GiftsSendContainer {...props1} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
it("should render GiftsSendContainer component", () => {
  wrapper = shallow(<GiftsSendContainer {...props2} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
it("should render GiftsSendContainer component", () => {
  wrapper = shallow(<GiftsSendContainer {...props3} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
it("should render GiftsSendContainer component", () => {
  wrapper = shallow(<GiftsSendContainer {...props2} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
it("should render GiftsSendContainer component", () => {
  wrapper = shallow(<GiftsSendContainer {...props4} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("should map state to props", () => {
  const initialState = {
    login: {
      loginStatus: true,
      detailsObject: {}
    },
    users: {
      cards: []
    }
  };
  expect(mapStateToProps(initialState)).toEqual({
    user: {},
    isLoggedIn: true,
    sentCards: []
  });
});
