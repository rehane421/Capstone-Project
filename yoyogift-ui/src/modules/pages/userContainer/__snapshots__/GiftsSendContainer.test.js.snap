// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GiftsSendContainer component 1`] = `
<WithStyles(ForwardRef(CircularProgress))
  style={
    Object {
      "marginLeft": "50%",
      "marginTop": "10%",
    }
  }
/>
`;

exports[`should render GiftsSendContainer component 2`] = `
<Redirect
  push={false}
  to="/"
/>
`;

exports[`should render GiftsSendContainer component 3`] = `
<h2
  style={
    Object {
      "background": "#b3d9f7",
      "color": "white",
      "fontSize": "20px",
      "fontWeight": "500",
      "height": "40px",
      "paddingTop": "5px",
      "textAlign": "center",
      "verticalAlign": "middle",
    }
  }
>
  NO DATA
</h2>
`;

exports[`should render GiftsSendContainer component 4`] = `
<Redirect
  push={false}
  to="/"
/>
`;

exports[`should render GiftsSendContainer component 5`] = `
<div>
  <WithStyles(GiftsSend)
    data={
      Array [
        Object {
          "id": 1,
        },
      ]
    }
  />
</div>
`;
